integration/tests/access/cohort3/access_store_tx_error_messages_test.go

```diff
@@ -104,37 +108,46 @@ func (s *AccessStoreTxErrorMessagesSuite) TestAccessStoreTxErrorMessages() {
 	txResult := s.createAndSendTxWithTxError()
 
-	txBlockID := convert.IDFromSDK(txResult.BlockID)
-	txID := convert.IDFromSDK(txResult.TransactionID)
-	expectedTxResultErrorMessage := txResult.Error.Error()
-
-	accessContainerName := "access_2"
-
 	// Wait until execution receipts are handled, transaction error messages are stored.
 	s.Eventually(func() bool {
-		value, err := s.getMaxReceiptHeight(accessContainerName)
+		value, err := s.getMaxReceiptHeight(s.accessContainerName)
 		return err == nil && value > txResult.BlockHeight
 	}, 60*time.Second, 1*time.Second)
 
 	// Stop the network containers before checking the results.
 	s.net.StopContainers()
 
-	// Get the access node and open the protocol DB.
-	accessNode := s.net.ContainerByName(accessContainerName)
-	// setup storage objects needed to get the execution data id
-	anDB, err := accessNode.DB()
-	require.NoError(s.T(), err, "could not open db")
+	txResults := []*sdk.TransactionResult{txResult}
+	txErrorMessages := s.fetchTxErrorMessages(txResults, s.accessContainerName)
+	s.verifyTxErrorMessage(txResults, txErrorMessages)
+}
 
-	metrics := metrics.NewNoopCollector()
-	anTxErrorMessages := badger.NewTransactionResultErrorMessages(metrics, anDB, badger.DefaultCacheSize)
+// TestBackfillTxErrorMessages verifies that transaction error messages are backfilled correctly
+// by creating a transaction that results in an error, running the backfill command, and checking
+// if the error message is stored and retrieved from the database.
+func (s *AccessStoreTxErrorMessagesSuite) TestBackfillTxErrorMessages() {
+	// Create and send a transactions that will result in an error.
+	transactionCount := 5
+	txResults := make([]*sdk.TransactionResult, transactionCount)
+	for i := 0; i < transactionCount; i++ {
+		txResults[i] = s.createAndSendTxWithTxError()
+	}
 
-	// Fetch the stored error message by block ID and transaction ID.
-	errMsgResult, err := anTxErrorMessages.ByBlockIDTransactionID(txBlockID, txID)
-	s.Require().NoError(err)
+	serverAddr := fmt.Sprintf("localhost:%s", s.net.ContainerByName(s.accessContainerName).Port(testnet.AdminPort))
+	adminClient := client.NewAdminClient(serverAddr)
+
+	startHeight := 1
+	endHeight := txResults[len(txResults)-1].BlockHeight // last tx result block height
+	data := map[string]interface{}{"start-height": startHeight, "endHeight": endHeight}
+	// executes the backfill command for transaction error messages
+	_, err := adminClient.RunCommand(context.Background(), "backfill-tx-error-messages", data)
+	require.NoError(s.T(), err)
+
+	// Stop the network containers before checking the results.
+	s.net.StopContainers()
 
-	// Verify that the error message retrieved matches the expected values.
-	s.Require().Equal(txID, errMsgResult.TransactionID)
-	s.Require().Equal(expectedTxResultErrorMessage, errMsgResult.ErrorMessage)
+	txErrorMessages := s.fetchTxErrorMessages(txResults, s.accessContainerName)
```

Comment:
how do you know that these weren't indexed regularly and were actually backfilled?

Reply:
we discussed, and do not see a clear way to address this. localnet testing shows that this is working, so I'm going to resolve this.